community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -458,10 +460,20 @@ private void commit() throws TransactionFailureException
 
         try ( CommitEvent commitEvent = transactionEvent.beginCommitEvent() )
         {
-            // Trigger transaction "before" hooks
-            if ( (hooksState = hooks.beforeCommit( txState, this, storeLayer )) != null && hooksState.failed() )
+            // Trigger transaction "before" hooks.
+            if ( hasChanges() )
```

Comment:
From a perspective of affected data fully agree, in the same time if you have a hook that lets say track last successful transaction commit time you will not be able to get that and in the same time your rollback hook will not gonna be invoked also. Which looks confusing. Do you agree?